js/src/stores/DbItemStore.js

```diff
@@ -18,149 +18,184 @@
 		ADD: "DbItemStore-ADD",
 		UPDATE: "DbItemStore-UDPATE",
 		DELETE: "DbItemStore-DELETE",
-			
+
 		STATE_VIEW: "DbItemStore-STATE_VIEW",
 		STATE_EDIT: "DbItemStore-STATE_EDIT"
-	}
+	};
 
 	var idSeq;
-	var dbItems = null;
-	
+	var dbItems = [];
+
 	var notify = pl.observable(DbItemStore);
-	
-	var getDbItemIndex = function(id) {		
+
+	var getDbItemIndex = function(id) {
 		return pl.findIndex(dbItems, function(dbItem) { return dbItem.db.id === id; });
 	};
-		
+
 	var updateDbItem = function(id, f) {
-	
+
 		var i = getDbItemIndex(id);
 		if ( i !== -1 ) {
 			var newDbItem = f( pl.extend({},dbItems[i]) );
 			dbItems = pl.update(dbItems, {$splice: [[i,1,newDbItem]]});
-		}				
+		}
 	};
-	
+
 	var load = function() {
-	
-		// FIXME - Need to load and also save to the userconfig, also the useconfig should be in another module
-		var userdbs = !fs.existsSync('userconfig.json') ? {} : JSON.parse( fs.readFileSync('userconfig.json', {encoding: 'utf8'}) );
-		var dbs = userdbs.databases || [];
-		
-		dbItems = dbs.map(function(db,i){	
-				
-			var mappedDb = pl.extend({},db);
-			mappedDb.id = i;
-		
-			return { state: DbItemStore.STATE_VIEW, db: mappedDb };
+
+		configApi.load().then(function(readConfig){
+			return readConfig;
+
+		}).catch(function(err){
+				console.log('Error while loading config. Initializing to empty.');
+				console.log(err);
+				return {};
+
+		}).then(function(readConfig){
+			config = readConfig;
+
+			var dbs = config.databases || [];
+
+			dbItems = dbs.map(function(db,i){
+				var mappedDb = pl.extend({},db);
+				mappedDb.id = i;
+
+				return { state: DbItemStore.STATE_VIEW, db: mappedDb };
+			});
+
+			idSeq = dbItems.length === 0 ? 0 : dbItems[dbItems.length-1].db.id + 1;
+			notify();
 		});
- 		
-		idSeq = dbItems.length == 0 ? 0 : dbItems[dbItems.length-1].db.id + 1;
-		
-		notify();
+
 	};
-	
+
 	var edit = function(id) {
-		
+
 		updateDbItem(id, function(dbItem) {
 			return pl.extend( dbItem, {state: DbItemStore.STATE_EDIT} );
 		});
-		
+
 		notify();
 	};
-	
+
 	var cancelEdit = function(id) {
-		
-		updateDbItem(id, function(dbItem) {
-			return pl.extend( dbItem, {state: DbItemStore.STATE_VIEW} );
-		});
-		
+
 		notify();
 	};
-	
+
 	var add = function(db) {
-		
+
 		db.id = idSeq++;
 
 		var newDbItem = { state: DbItemStore.STATE_VIEW, db: db };
-	
+
 		dbItems = pl.update(dbItems, {$push: [newDbItem]});
-		
+
+		saveConfiguration();
+
 		notify();
 	};
-	
+
 	var update = function(db) {
-		
+
 		updateDbItem(db.id, function(dbItem) {
 			return pl.extend( dbItem, {state: DbItemStore.STATE_VIEW, db: db} );
 		});
-		
+
+		saveConfiguration();
+
 		notify();
 	};
-	
+
+	function saveConfiguration() {
+
+		dbConfig = dbItems.map(function(dbItem){
+			return dbItem.db;
+		});
+
+		config.databases = dbConfig;
+
+		configApi.save(config).catch(function(err){
+			alert('Error while saving cofiguration');
+			console.log('Error while saving configuration:');
+			console.log(err);
+		});
```

Comment:
I don't like the "alert" here. This should be done in the views later on.

Use TAG when logging,

console.log(TAG, thing_to_log );